File: src/main/java/com/elchologamer/userlogin/listener/restriction/ChatRestriction.java
package com.elchologamer.userlogin.listener.restriction;

import com.elchologamer.userlogin.ULPlayer;
import org.bukkit.event.EventHandler;
import org.bukkit.event.player.AsyncPlayerChatEvent;

public class ChatRestriction extends BaseRestriction<AsyncPlayerChatEvent> {

    public ChatRestriction() {
        super("chat");
    }

    @EventHandler
    public void handle(AsyncPlayerChatEvent e) {
        if (!shouldRestrict(e)) return;

        e.setCancelled(true);
        ULPlayer.get(e.getPlayer()).sendMessage("messages.chat_disabled");
    }
}

File: src/main/kotlin/com/elchologamer/userlogin/listener/restriction/ChatRestriction.kt
package com.elchologamer.userlogin.listener.restriction

import com.elchologamer.userlogin.ULPlayer
import org.bukkit.event.EventHandler
import org.bukkit.event.player.AsyncPlayerChatEvent

object ChatRestriction : BaseRestriction<AsyncPlayerChatEvent>("chat") {
    @EventHandler
    fun handle(e: AsyncPlayerChatEvent) {
        if (!shouldRestrict(e)) return
        e.isCancelled = true
        ULPlayer[e.player].sendMessage("messages.chat_disabled")
    }
}